ecl/hqlcpp/hqlttcpp.cpp

```diff
@@ -11310,6 +11310,27 @@ IHqlExpression * AnnotationNormalizerTransformer::createTransformed(IHqlExpressi
     return queryLocationIndependentExtra(body)->cloneAnnotations(transformed);
 }
 
+void AnnotationNormalizerTransformer::setTransformed(IHqlExpression * expr, IHqlExpression * transformed)
+{
+    NewHqlTransformer::setTransformed(expr, transformed);
+
+    //All derived expressions map to the same transformed expressions => override the default behaviour in
+    //NewHqlTransformer::transform which uses transformed->queryBody(true) for body
+    IHqlExpression * body = expr->queryBody(true);
+    if (body != expr)
```

Comment:
Isn't this while(expr != body) ? Ind would it be clearer coded as such?

Reply:
yes it would.  The original code which I copied and modified would also be slightly better recoded.  I will revisit.